x/ccv/provider/types/msg.go

```diff
@@ -83,167 +71,215 @@ func NewMsgAssignConsumerKey(consumerId string, providerValidatorAddress sdk.Val
 	}, nil
 }
 
-// Route implements the sdk.Msg interface.
-func (msg MsgAssignConsumerKey) Route() string { return RouterKey }
-
-// Type implements the sdk.Msg interface.
-func (msg MsgAssignConsumerKey) Type() string {
-	return TypeMsgAssignConsumerKey
-}
-
-// GetSigners implements the sdk.Msg interface. It returns the address(es) that
-// must sign over msg.GetSignBytes().
-// If the validator address is not same as delegator's, then the validator must
-// sign the msg as well.
-func (msg MsgAssignConsumerKey) GetSigners() []sdk.AccAddress {
```

Comment:
Why are the `GetSigners`, `GetSignBytes`, etc. removed? Is it a new cosmos-sdk thing were they're not needed anymore? If yes, how do we verify that, for example `MsgAssignConsumerKey` is signed by `msg.ProviderAddr`?

Reply:
They are no longer needed in SDK 0.50 (see [UPGRADING.md](https://github.com/cosmos/cosmos-sdk/blob/v0.50.9-lsm/UPGRADING.md)):
> GetSigners() is no longer required to be implemented on Msg types. The SDK will automatically infer the signers from the Signer field on the message. The signer field is required on all messages unless using a custom signer function.

> Messages no longer need to implement the LegacyMsg interface and implementations of GetSignBytes can be deleted. Because of this change, global legacy Amino codec definitions and their registration in init() can safely be removed as well.

> how do we verify that, for example MsgAssignConsumerKey is signed by msg.ProviderAddr

Why do we have both signer and provider_addr in MsgAssignConsumerKey?

```protobuf
message MsgAssignConsumerKey {
  option (cosmos.msg.v1.signer) = "signer";

  // The validator address on the provider
  string provider_addr = 2 [ (gogoproto.moretags) = "yaml:\"address\"" ];

  // Tx signer address
  string signer = 4 [(cosmos_proto.scalar) = "cosmos.AddressString"];
}
```